sdks/python/apache_beam/runners/interactive/pipeline_instrument.py

```diff
@@ -375,12 +377,15 @@ def _read_cache(self, pipeline, pcoll):
     # Can only read from cache when the cache with expected key exists.
     if self._cache_manager.exists('full', key):
       if key not in self._cached_pcoll_read:
+        if pcoll.is_bounded:
+          read_transform = cache.ReadCache(self._cache_manager, key)
+        else:
+          read_transform = streaming_cache.StreamingCacheReceiver()
```

Comment:
Should we unify both bounded and unbounded pcollections to use the streaming cache?

Reply:
This boundedness is implemented 4 months ago in this PR: https://github.com/apache/beam/commit/79d4bf28c78ba8b7e9a0d01f6c86db79868312b9
Before, Python SDK doesn't care about the boundedness of pvalue instances.

It simply checks the boundedness of io source and pass along the information: https://github.com/apache/beam/commit/79d4bf28c78ba8b7e9a0d01f6c86db79868312b9#diff-5bdf3a832591550d5839cb5f76ea2f10R884

So it cannot be source of truth for checking PCollection boundedness.

This can be another motivation for us to unify the 2 caching managers.

Reply:
The is_bounded is the source of truth for checking its boundedness. If you look at the other streaming sources in that linked PR, it shows them creating the PCollection with is_bounded=False.

Reply:
Yes, I think this works for most scenarios. Do you remember I used to check the source `is_bounded` field when checking if a pipeline has unbounded source? Then we switched to check if a pipeline contains known unbounded sources.
It's the same thing here.
Unless we change the implementation of `is_bounded` to cover all possible sources, we can never be sure if the `is_bounded` is the source of truth. 

We have these questions to answer:

- Does all source implementation inherit the `apache_beam/io/iobase.py`?

- And even if they do, does all the unbounded source implementation put a `True` value in the `is_bounded` member in their source code? Is there any enforcement for that except code review?

- Can there be unbounded source dynamically created without source code?

I think this is equivalent to checking if a given pipeline has unbounded source.
If we cannot be sure there (for example, user-defined unknown sources with arbitrary definition) since we only check known unbounded sources, we cannot be sure here either.